main.nf

```diff
@@ -464,134 +519,205 @@ dwi_for_test_gibbs
 
 process Topup {
-    cpus 2
+    cpus 4
 
     input:
-    set sid, file(dwi), file(bval), file(bvec), file(rev_b0), readout, encoding\
-        from dwi_gradients_rev_b0_for_topup
+      set sid, file(rev_b0), file(b0),  readout, encoding\
+        from rev_b0_with_readout_encoding_for_topup
 
     output:
-    set sid, "${sid}__corrected_b0s.nii.gz", "${params.prefix_topup}_fieldcoef.nii.gz",
-    "${params.prefix_topup}_movpar.txt" into topup_files_for_eddy_topup
-    file "${sid}__rev_b0_warped.nii.gz"
-    file "${sid}__b0_mean.nii.gz"
+      set sid, "${sid}__corrected_b0s.nii.gz", "${params.prefix_topup}_fieldcoef.nii.gz",
+      "${params.prefix_topup}_movpar.txt" into topup_files_for_eddy_topup
+      file "${sid}__rev_b0_warped.nii.gz"
 
     when:
-    params.run_topup && params.run_eddy
+      params.run_topup && params.run_eddy
 
     script:
     """
-    export OMP_NUM_THREADS=$task.cpus
-    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
-    export OPENBLAS_NUM_THREADS=1
-    export ANTS_RANDOM_SEED=1234
-    scil_extract_b0.py $dwi $bval $bvec ${sid}__b0_mean.nii.gz --mean\
-        --b0_thr $params.b0_thr_extract_b0 --force_b0_threshold
-    scil_image_math.py mean $rev_b0 $rev_b0 --data_type float32 -f
-    antsRegistrationSyNQuick.sh -d 3 -f ${sid}__b0_mean.nii.gz -m $rev_b0 -o output -t r -e 1
-    mv outputWarped.nii.gz ${sid}__rev_b0_warped.nii.gz
-    scil_prepare_topup_command.py ${sid}__b0_mean.nii.gz ${sid}__rev_b0_warped.nii.gz\
-        --config $params.config_topup\
-        --encoding_direction $encoding\
-        --readout $readout --out_prefix $params.prefix_topup\
-        --out_script
-    sh topup.sh
-    cp corrected_b0s.nii.gz ${sid}__corrected_b0s.nii.gz
+      export OMP_NUM_THREADS=$task.cpus
+      export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+      export OPENBLAS_NUM_THREADS=1
+      export ANTS_RANDOM_SEED=1234
+      antsRegistrationSyNQuick.sh -d 3 -f $b0 -m $rev_b0 -o output -t r -e 1
+      mv outputWarped.nii.gz ${sid}__rev_b0_warped.nii.gz
+      scil_prepare_topup_command.py $b0 ${sid}__rev_b0_warped.nii.gz\
+          --config $params.config_topup\
+          --encoding_direction $encoding\
+          --readout $readout --out_prefix $params.prefix_topup\
+          --out_script
+      sh topup.sh
+      cp corrected_b0s.nii.gz ${sid}__corrected_b0s.nii.gz
     """
 }
 
-dwi_for_eddy
-    .join(gradients_for_eddy)
-    .join(b0_mask_for_eddy)
-    .join(readout_encoding_for_eddy)
-    .set{dwi_gradients_mask_topup_files_for_eddy}
+dwi_for_eddy_topup.into{complex_dwi_for_eddy_topup;simple_dwi_for_eddy_topup}
 
-process Eddy {
+// DATA FOR CONCATENATE
+complex_dwi_for_eddy_topup.combine(sid_rev_dwi_included.collect().toList())
+        .filter{ (it[0] in it[3]) }
+        .map{it -> [it[0], it[1]]}
+        .set{dwi_for_prepare_for_eddy}
+
+// DATA DO NOT CONCATENATE
+expl1 = Channel.value(0)
+
+simple_dwi_for_eddy_topup.combine(sid_rev_dwi_excluded.collect().toList())
```

Comment:
This is not used anywhere else, but I think I know where you want to go with it. We need to talk about it, but long story short, this is useless if we keep the current behavior where if inputs have different reverse acquisition specification, we don't run the pipeline